song-server/src/test/java/bio/overture/song/server/service/AnalysisServiceSenderTest.java

```diff
@@ -45,66 +53,92 @@
 
   private String analysisId;
 
+  private AnalysisGenerator analysisGenerator;
+  private Analysis analysis;
+
   @Before
   public void beforeTest() {
     this.studyId = RANDOM_GENERATOR.generateRandomAsciiString(10);
     this.analysisId = RANDOM_GENERATOR.generateRandomUUIDAsString();
+
+    analysisGenerator = createAnalysisGenerator(studyId, internalAnalysisService, RANDOM_GENERATOR);
+    val analysisSchema =
+        AnalysisSchema.builder()
+            .name("test-schema")
+            .version(1)
+            .schema(new ObjectMapper().createObjectNode())
+            .build();
+    val analysisData = AnalysisData.builder().data(new ObjectMapper().createObjectNode()).build();
+    this.analysis =
+        Analysis.builder()
+            .analysisId(this.analysisId)
+            .studyId(this.studyId)
+            .analysisState(UNPUBLISHED.name())
+            .analysisSchema(analysisSchema)
+            .analysisData(analysisData)
+            .build();
     this.studyId = RANDOM_GENERATOR.generateRandomAsciiString(15);
   }
 
   @Test
   public void testAnalysisCreate() {
     when(internalAnalysisService.create(studyId, DUMMY_PAYLOAD)).thenReturn(analysisId);
-    val analysisServiceSender = createTestAnalysisServiceSender(UNPUBLISHED);
+    when(internalAnalysisService.unsecuredDeepRead(analysisId)).thenReturn(this.analysis);
```

Comment:
if the previous comment for the partial refactor is done, then mocking of the unsecuredDeepRead is not needed at all